arch/arc/core/fatal.c

```diff
@@ -84,3 +84,10 @@ FUNC_NORETURN void _NanoFatalErrorHandler(unsigned int reason,
 	for (;;)
 		;
 }
+
+
+FUNC_NORETURN void _arch_syscall_oops(void *ssf_ptr)
+{
+	_SysFatalErrorHandler(_NANO_ERR_KERNEL_OOPS, ssf_ptr);
```

Comment:
Are you sure that ssf_ptr is laid out exactly like a NANO_ESF?
On x86 it wasn't, I had to copy some stuff into a new data structure.

Reply:
In ARC, for syscall_oops, the ssf_ptr has no meaning. It just be there to let compile ok.  ssf_ptr is designed to point to the exception stack. In current ARC's codes,  it's not the case. I will modify the codes related to ssf_ptr as a whole in the future.

Reply:
OK. What we want when we call _SysFatalErrorHandler is for the provided NANO_ESF parameter to contain the PC/registers when the system call was invoked. At least the PC, so the user can figure out what syscall invocation from their application code caused the fault.